src/code/z_debug.c

```diff
@@ -1,20 +1,327 @@
 #include <ultra64.h>
 #include <global.h>
+#include <z64.h>
+#include <color.h>
+#include <regs.h>
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_debug/func_800636C0.s")
+#define BUTTON_C_RIGHT        0x0001
+#define BUTTON_C_LEFT         0x0002
+#define BUTTON_C_DOWN         0x0004
+#define BUTTON_C_UP           0x0008
+#define BUTTON_R              0x0010
+#define BUTTON_L              0x0020
+#define BUTTON_D_RIGHT        0x0100
+#define BUTTON_D_LEFT         0x0200
+#define BUTTON_D_DOWN         0x0400
+#define BUTTON_D_UP           0x0800
+#define BUTTON_START          0x1000
+#define BUTTON_Z              0x2000
+#define BUTTON_B              0x4000
+#define BUTTON_A              0x8000
 
-void func_8006375C(UNK_TYPE arg0, UNK_TYPE arg1, UNK_TYPE arg2)
+#define REG_GROUP_TOTAL 29
+#define REG_N 96
+#define PAGE_N 16
```

Comment:
I don't think these defines belong in this file since they are/will be relevant in other places.

Reply:
PAGE_N i expect to be specific to z_debug.c. the rest i moved to reg.h

Reply:
Shouldn't `REG_N` actually be something like `PAGE_NUM * PAGE_SIZE`?

Reply:
`REG_GROUP_TOTAL` became `REG_GROUPS`, and `REG_N became `REG_PER_GROUP` in my last commit, such that the size of the data array is now defined as `REG_GROUPS * REG_PER_GROUP`

Reply:
Yeah I think that's fine, but my point is I doubt they defined the size of a group, and instead defined the size of a page and a number of pages per group. Or at least that's how I would do it. Unless I'm missing something hinting otherwise.